roles/azure_virtual_machine_with_public_ip/tasks/create.yml

```diff
@@ -41,10 +41,36 @@
         resource_group: "{{ azure_virtual_machine_with_public_ip_resource_group }}"
       register: vnet_info
 
-    - name: Ensure vnet exists
-      ansible.builtin.fail:
-        msg: When no network interfaces are specified, a virtual network must exist
+    - name: Create virtual network if not already exist
       when: vnet_info.virtualnetworks | length == 0
+      block:
+        - name: Check virtual network CIDR address prefixes
+          ansible.builtin.fail:
+            msg: Azure virtual network CIDR address prefixes must be defined as azure_virtual_machine_with_public_ip_vm.vnet_address_prefixes_cidr
+          when: azure_virtual_machine_with_public_ip_vm.vnet_address_prefixes_cidr is not defined
+
+        - name: Check subnet CIDR address prefixes
+          ansible.builtin.fail:
+            msg: Azure subnet CIDR address prefixes must be defined as azure_virtual_machine_with_public_ip_vm.subnet_address_prefixes_cidr
+          when: azure_virtual_machine_with_public_ip_vm.subnet_address_prefixes_cidr is not defined
+
+        - name: Create virtual network and subnet
+          ansible.builtin.include_role:
+            name: cloud.azure_ops.azure_manage_networking_stack
+          vars:
+            azure_manage_networking_stack_operation: create
+            azure_manage_networking_stack_resource_group: "{{ azure_virtual_machine_with_public_ip_resource_group }}"
+            azure_manage_networking_stack_virtual_network: "{{ vm_name }}"
+            azure_manage_networking_stack_subnet: "{{ vm_name }}"
+            azure_manage_networking_stack_vnet_address_prefixes_cidr: "{{ azure_virtual_machine_with_public_ip_vm.vnet_address_prefixes_cidr }}"
+            azure_manage_networking_stack_subnet_address_prefixes_cidr: "{{ azure_virtual_machine_with_public_ip_vm.subnet_address_prefixes_cidr }}"
+            azure_manage_networking_stack_tags:
+              _own_vnet_: "{{ vm_name }}"
+
+    - name: Refresh virtual network info
```

Comment:
When it is part of the block above, in case there is no need to create a networking stack and the block above skipped, vnet_info doesn't contain real information, just a message that this task was skipped.

Reply:
Oh wow.  Ansible really does that?  If the task is skipped it will populate vnet_info with nothing!?  I would have expected a skip task to not populate the register variable.